core/include/kernel/mutex_pm_aware.h

```diff
@@ -0,0 +1,40 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2024, STMicroelectronics
+ */
+#ifndef __KERNEL_MUTEX_PM_AWARE_H
+#define __KERNEL_MUTEX_PM_AWARE_H
+
+#include <kernel/mutex.h>
+#include <kernel/spinlock.h>
+
+/*
+ * struct mutex_pm_aware - Mutex usable in PM atomic sequence
+ *
+ * Some resources need a mutex protection for runtime operations but are
+ * also accessed during specific system power transition (PM power off,
+ * suspend and resume) that operate in atomic execution environment where
+ * non-secure world is not operational, for example in fastcall SMC entries
+ * of the PSCI services. In such case we cannot take a mutex and we expect
+ * the mutex is unlocked. Additionally a spinning lock is attempted to be
+ * locked to check the resource access consistency. In case of contention
+ * between thread and non-thread access to the resource, core will
```

Comment:
Contention of non-threaded access always results in a panic. Only contention between threaded accesses will be handled gracefully. This description needs to be rephrased to make that clear.

Please split the comment into two paragraphs, one with the background and one short that describes the intended behavior.

Reply:
Ok, I understand my phrasing is confusing. I'll do.